x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/vectortile/AbstractVectorTileSearchAction.java

```diff
@@ -226,9 +234,31 @@ public void setSize(int size) {
         }
 
         public void setAggBuilder(AggregatorFactories.Builder aggBuilder) {
-            // TODO: validation
+            for (AggregationBuilder aggregation : aggBuilder.getAggregatorFactories()) {
```

Comment:
Let's wait with this. I think I have some idea on how to make this more generic.

Reply:
I still wonder if 1) this should responsibility of the callers and 2) if we want to add these pipeline aggs, we need to throw an exception if we don't recognize the agg

Reply:
My hope is that a user of this API does not need to be an expert on Elasticsearch so we can provide this information from them. I am hoping to only support a subset of the aggregations, the ones that make sense.